packages/fast-usdc/src/exos/status-manager.js

```diff
@@ -1,12 +1,171 @@
+import { M } from '@endo/patterns';
+import { makeError, q } from '@endo/errors';
+
+import { appendToStoredArray } from '@agoric/store/src/stores/store-utils.js';
+import { CctpTxEvidenceShape, PendingTxShape } from '../typeGuards.js';
+import { TxStatus } from '../constants.js';
+
 /**
+ * @import {MapStore, SetStore} from '@agoric/store';
  * @import {Zone} from '@agoric/zone';
+ * @import {CctpTxEvidence, NobleAddress, SeenTxKey, PendingTxKey, PendingTx} from '../types.js';
+ */
+
+/**
+ * Create the key for the pendingTxs MapStore
+ * @param {NobleAddress} addr
+ * @param {bigint} amount
+ */
+const toPendingTxKey = (addr, amount) =>
+  /** @type {PendingTxKey} */ (JSON.stringify([addr, String(amount)]));
+
+/**
+ * Get the key for the pendingTxs MapStore
+ * @param {CctpTxEvidence} evidence
+ * @returns {PendingTxKey}
+ */
+const getPendingTxKey = evidence => {
+  const { amount, forwardingAddress } = evidence.tx;
+  return toPendingTxKey(forwardingAddress, amount);
+};
+
+/**
+ * Get the key for the seenTxs SetStore
+ * @param {CctpTxEvidence} evidence
  */
+const getSeenKey = evidence => {
+  const { txHash, chainId } = evidence;
+  return /** @type {SeenTxKey} */ (JSON.stringify([txHash, chainId]));
+};
 
 /**
+ * The `StatusManager` keeps track of Pending and Seen Transactions
+ * via {@link TxStatus} states, aiding in coordination between the `Advancer`
+ * and `Settler`.
+ *
+ * XXX consider separate facets for `Advancing` and `Settling` capabilities.
+ *
  * @param {Zone} zone
  */
 export const prepareStatusManager = zone => {
-  return zone.exo('Fast USDC Status Manager', undefined, {});
+  /** @type {MapStore<PendingTxKey, PendingTx[]>} */
+  const pendingTxs = zone.mapStore('PendingTxs', {
+    keyShape: M.string(),
+    valueShape: M.arrayOf(PendingTxShape),
+  });
+
+  /** @type {SetStore<SeenTxKey>} */
+  const seenTxs = zone.setStore('SeenTxs', {
+    keyShape: M.string(),
+  });
+
+  /**
+   * Ensures that `txHash+chainId` has not been processed
+   * and adds `txHash+chainId` to `seenTxs` set.
+   *
+   * @param {CctpTxEvidence} evidence
+   */
+  const assertNotSeen = evidence => {
+    const seenKey = getSeenKey(evidence);
+    if (seenTxs.has(seenKey)) {
+      throw makeError(`Transaction already seen: ${q(seenKey)}`);
+    }
+    seenTxs.add(seenKey);
+  };
+
+  return zone.exo(
+    'Fast USDC Status Manager',
+    M.interface('StatusManagerI', {
+      advance: M.call(CctpTxEvidenceShape).returns(M.undefined()),
+      observe: M.call(CctpTxEvidenceShape).returns(M.undefined()),
+      hasPendingSettlement: M.call(M.string(), M.bigint()).returns(M.boolean()),
+      settle: M.call(M.string(), M.bigint()).returns(M.undefined()),
+      view: M.call(M.string(), M.bigint()).returns(M.arrayOf(PendingTxShape)),
```

Comment:
this is a new name for what is usually called `lookup`. It also reads as if it could be a state transition.  Please change.